generators/dart.js

```diff
@@ -66,12 +66,19 @@ Blockly.Dart.ORDER_RELATIONAL = 9;     // >= > <= < as is is!
 Blockly.Dart.ORDER_EQUALITY = 10;      // == !=
 Blockly.Dart.ORDER_LOGICAL_AND = 11;   // &&
 Blockly.Dart.ORDER_LOGICAL_OR = 12;    // ||
-Blockly.Dart.ORDER_CONDITIONAL = 13;   // expr ? expr : expr
-Blockly.Dart.ORDER_CASCADE = 14;       // ..
-Blockly.Dart.ORDER_ASSIGNMENT = 15;    // = *= /= ~/= %= += -= <<= >>= &= ^= |=
+Blockly.Dart.ORDER_IF_NULL = 13;       // ??
+Blockly.Dart.ORDER_CONDITIONAL = 14;   // expr ? expr : expr
+Blockly.Dart.ORDER_CASCADE = 15;       // ..
+Blockly.Dart.ORDER_ASSIGNMENT = 16;    // = *= /= ~/= %= += -= <<= >>= &= ^= |=
 Blockly.Dart.ORDER_NONE = 99;          // (...)
 
 /**
+ * Allow for switching between one and zero based indexing, one based by
```

Comment:
I'd expand this comment a bit (same for other languages).
...based indexing for lists and text, ...